dockstore-common/src/main/java/io/dockstore/common/yaml/constraints/ValidPrimaryPathValidator.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dockstore.common.yaml.constraints;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+/**
+ * Validates the primary descriptor's file path.
+ */
+public class ValidPrimaryPathValidator implements ConstraintValidator<ValidPrimaryPath, String> {
+
+    @Override
+    public boolean isValid(final String path, final ConstraintValidatorContext context) {
+        // nulls are valid, see note in BaseConstraintValidator for more.
+        return path.isEmpty() || path.startsWith("/");
```

Comment:
Not sure I understand the comment. If nulls are valid, is there a chance of the methods on here dying with a `NullPointerException`, should there be a null check like in `ValidTestFilePathsValidator`, etc.?

Reply:
Oops, sorry this comment shouldn't be here. I was copying an existing validator and forgot to remove this...